service/history/task/cross_cluster_target_task_executor.go

```diff
@@ -222,6 +227,105 @@ func (t *crossClusterTargetTaskExecutor) executeCancelExecutionTask(
 	return &types.CrossClusterCancelExecutionResponseAttributes{}, nil
 }
 
+func (t *crossClusterTargetTaskExecutor) executeApplyParentClosePolicyTask(
+	ctx context.Context,
+	task *crossClusterTargetTask,
+) (*types.CrossClusterApplyParentClosePolicyResponseAttributes, error) {
+	if task.processingState != processingStateInitialized {
+		return nil, errUnknownTaskProcessingState
+	}
+
+	attributes := task.request.ApplyParentClosePolicyAttributes
+	if attributes == nil {
+		return nil, errMissingTaskRequestAttributes
+	}
+
+	for _, childAttrs := range attributes.AppyParentClosePolicyAttributes {
+		targetDomainName, err := t.verifyDomainActive(childAttrs.ChildDomainID)
+		if err != nil {
+			return nil, err
+		}
+
+		scope := t.metricsClient.Scope(metrics.CrossClusterTaskTypeApplyParentClosePolicyScope)
+		err = applyParentClosePolicy(
+			ctx,
+			t.historyClient,
+			childAttrs.ChildDomainID,
+			targetDomainName,
+			childAttrs.ChildWorkflowID,
+			childAttrs.ChildRunID,
+			*childAttrs.ParentClosePolicy,
+		)
+		switch err.(type) {
+		case *types.EntityNotExistsError, *types.WorkflowExecutionAlreadyCompletedError, *types.CancellationAlreadyRequestedError:
+			// expected error, no-op
+			break
+		default:
+			scope.IncCounter(metrics.ParentClosePolicyProcessorFailures)
+			return nil, err
+		}
+		scope.IncCounter(metrics.ParentClosePolicyProcessorSuccess)
+	}
+
+	// TODO: Consider going through all the children, even if some fail to apply the parent policy,
+	// and add the policy application status by a child identifier (domain-wf-run id)
+	// Right now, return value is all or none even if we were able apply the policy on some children successfuly
+	return &types.CrossClusterApplyParentClosePolicyResponseAttributes{}, nil
+}
+
+func (t *crossClusterTargetTaskExecutor) executeRecordChildWorkflowExecutionCompleteTask(
+	ctx context.Context,
+	task *crossClusterTargetTask,
+) (*types.CrossClusterRecordChildWorkflowExecutionCompleteResponseAttributes, error) {
+	if task.processingState != processingStateInitialized {
+		return nil, errUnknownTaskProcessingState
+	}
+
+	// parent info is in attributes
+	attributes := task.request.RecordChildWorkflowExecutionCompleteAttributes
+	if attributes == nil {
+		return nil, errMissingTaskRequestAttributes
+	}
+
+	_, err := t.verifyDomainActive(attributes.TargetDomainID)
+	if err != nil {
+		return nil, err
+	}
+
+	// child info is in taskInfo
+	taskInfo, ok := task.Info.(*persistence.CrossClusterTaskInfo)
+	if !ok {
+		return nil, errUnexpectedTask
+	}
+
+	switch task.processingState {
```

Comment:
nit: we can remove this switch since we've verified the processingState on L280?